test/Kestrel.Core.Tests/Http2ConnectionTests.cs

```diff
@@ -2061,6 +2061,17 @@ public Task HEADERS_Received_HeaderBlockContainsTEHeader_ValueIsNotTrailers_Stre
 
             await SendGoAwayAsync();
 
+            var spinLimit = 50; // 2.5 seconds
+            while (_connection.State == Http2Connection.ConnectionState.Open && spinLimit --> 0)
+            {
+                await Task.Delay(50);
+            }
+            Assert.Equal(Http2Connection.ConnectionState.Closing, _connection.State);
+
+            await SendRstStreamAsync(1);
```

Comment:
Change this test to use graceful request completions. E.g. use a different app that waits for data before responding, then send a final data frame after the go away.